traits/src/fee.rs

```diff
@@ -17,6 +17,12 @@ pub trait SwappablePaymentAssetTrader<AccountId, AssetId, Amount>: InspectTransa
 		asset_out_amount: Amount,
 	) -> Result<Amount, DispatchError>;
 
+	fn calculate_out_given_in(
+		asset_in: AssetId,
+		asset_out: AssetId,
+		asset_in_amount: Amount,
```

Comment:
nitpick: the generic here should Balance - amount is considered signed ( check primitives::Amount type). 

just to be consistent.